build-tools/update

```diff
@@ -37,10 +37,19 @@ try {
 } catch (e) {}
 
 let configBuf = fs.readFileSync('config/config.js', {encoding: 'utf8'});
-configBuf = configBuf.replace(/\/\* version \*\/[^;\n]*;/, `/* version */ Config.version = ${JSON.stringify(version)};`);
+const routes = JSON.parse(fs.readFileSync('config/routes.json'));
+configBuf = configBuf.replace(/\/\* version \*\/[^;\n]*;/, `/* version */ Config.version = ${JSON.stringify(version)};`)
+	.replace(/\/\* root \*\/[^,\n]*,/, `/* root */ '${routes.root}',`)
+	.replace(/\/\* client \*\/[^,\n]*,/, `/* client */ '${routes.client}',`)
+	.replace(/\/\* dex \*\/[^,\n]*,/, `/* dex */ '${routes.dex}',`)
+	.replace(/\/\* replays \*\/[^,\n]*,/, `/* replays */ '${routes.replays}',`)
+	.replace(/\/\* users \*\/[^,\n]*,/, `/* users */ '${routes.users}',`);
 
 fs.writeFileSync('config/config.js', configBuf);
 console.log("DONE");
+const context = {Config: {}};
+require('vm').runInNewContext(configBuf, context);
+const Config = context.Config;
```

Comment:
Neat hack. Not sure if I like this, but I suppose there's no better way.

Reply:
Wait, I'm WTFing at this as well. At the very least, some comments would help. AIUI you:

1) read in the routes json
2) create an automatically generated config fragment.
3) try to read in the config file (and remove an old auto generated config comments) and append the config fragment with routes at the end
4) write this config

- this is all to workaround the fact that the routes.json cant be loaded in the browser so you need to embedded it in the config.json (OK)
- this is also somewhat fragile if a config already has routes defined and we trample over them by appending ours at the end to take precedence (probably not super concerning)
- these sorts of manipulations become much easier if the config is a *data* specification encoding like JSON/JSON5 instead of full blown JS (out of scope)

5) We eval the config (this is in fact just eval, no? Why bother with the vm?).

Once again, much less hacky if the config were JSON and not JS, but thats out of scope. **Why do we need to read the config in again though?** We already know what the value for routes has to be (because we append our autoconfig at the end anyway, so we know Config.routes  is going to equal what we read from routes.json, ie the routes constant)?

Possibly missing something (which means this section could benefit from some comments explaining whatever misconception I have) :)

Reply:
Oh, evaling the config was a leftover hack from when the routes were actually defined in the config file. I can remove that now.

Reply:
I agree that moving the modifiable configuration to a JSON file and generating a js file from that would be better, but imo that would go in its own PR.